Automattic/jetpack
# Voice to Content: Fix audio recording where WebM is not supported (iOS for example)

Fixes #36159.

## Proposed changes:

* Check support for `mp4` with `mp4a` and use it when available (iOS for example)
* Default to `webm` when `mp4` is not supported (the format we used before)
* Make sure we set the media type when creating the MediaRecorder instance, not just when generating the respective blobs

### Other information:


## Jetpack product discussion



## Does this pull request change what data or activity we track or use?



## Testing instructions:






* Go to the block editor of the provided test site, insert a Voice to Content block and record an audio by clicking on "Begin recording"
* Wait for the transcription to be ready; the block will be closed automatically and the transcription will be present on the editor
* Test it on:
   * Desktop with multiple OS and browser
   * Mobile with multiple OS and browser
   * It should work correctly on all platforms
